apps/shared/avifpng.c

```diff
@@ -462,6 +566,9 @@ avifBool avifPNGWrite(const char * outputFilename, const avifImage * avif, uint3
             rowPointers[y] = &rgb.pixels[y * rgb.rowBytes];
         }
     }
+    if (avifImageGetExifOrientationFromIrotImir(avif) != 1) {
+        // TODO(yguyon): Rotate the samples.
```

Comment:
..or also some day construct a minimal Exif like for JPEGs - as PNG is also a "compressed" format, the mandatory tags are the same (see also overlap w/ the [pHYs chunk](https://www.w3.org/TR/png/#11pHYs)).

OTOH, there is no guarantee that all 3rd party SW will access and use the Exif info in a PNG...

Reply:
According to the Extensions to the PNG 1.2 Specification, Version 1.5.0, section 3.7:

> It is recommended that unless a decoder has independent knowledge of the validity of the Exif data, the data should be considered to be of historical value only.

So Exif orientation should *not* be taken into account when displaying a PNG.

Reply:
Ah, good find.

So I guess this means you always (if irot/imir not 1) want to rotate the PNG output, and you always want to reset Exif Orientation to 1 (if prior Exif data exists)?

See also https://zpl.fi/exif-orientation-in-different-formats/

Reply:
Yes.